### ISSUE TITLE
Docker pre-submit checks

### ISSUE THREAD

**Issue by akirabaruah** [cardinalitypuzzles/smallboard#223]
To ensure that the Dockerized environment stays valid, we should enforce pre-submit checks to:

- build the smallboard docker image from the `Dockerfile`
- run the server `docker-compose` as a sanity check

**Comment by rgossiaux** [comment]
Going to wait until we do #228 to do this.

**Comment by cs-miller** [comment]
wondering if we have a limited number of docker pulls and/or github actions minutes

**Comment by rgossiaux** [comment]
We're not using docker hub (the container is just for dev right now though we could deploy it potentially if we wanted to or switched infra) but there are github action limits. I doubt we'll hit them though.

**Comment by cs-miller** [comment]
yeah i vaguely remembered they were introducing[ some pull limits](https://www.docker.com/increase-rate-limits) recently, but should be non-issue
